parlai/tasks/coqa/build.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+# Download and build the data if it does not exist.
+
+import parlai.core.build_data as build_data
+import os
+import json
+
+
+def make_parlai_format(outpath, dtype, data):
+    print('building parlai:' + dtype)
+    fout = open(os.path.join(outpath, dtype + '.txt'), 'w')
+    for each in data:
+        s = ""
```

Comment:
If you've ever used a StringBuilder in Java, python has similar semantics (strings are immutable).

So you need to collect into a list:
```
output = []
output.append('text:')
output.append(...)
outstr = "".join(output)
```

However, for this particular piece of code, I think things will be more readable with a format string, something akin to::

```python
output = "text:{question}\tlabels:{labels}".format(
    question=question_text,
    labels=labels_text
)
```

Reply:
Thanks for that! That is very useful!